iocore/net/quic/QUICFrameGenerator.cc

```diff
@@ -24,9 +24,9 @@
 #include "QUICFrameGenerator.h"
 
 void
-QUICFrameGenerator::_records_frame(QUICFrameId id, QUICFrameInformation info)
+QUICFrameGenerator::_records_frame(QUICFrameId id, QUICFrameInformationUPtr &info)
 {
-  this->_info.insert(std::make_pair(id, info));
+  this->_info.insert(std::make_pair(id, std::move(info)));
```

Comment:
Why don't you receive `QUICFrameInformationUPtr` but not a reference for it.
`_records_frame(QUICFrameId id, QUICFrameInformationUPtr info)`

A caller of `_records_frame` wouldn't expect that it loses the ownership of `info`.

Reply:
Because I'd prefer to reducing the call of std::move. 

But I'm ok with that. Will change.

Reply:
On second thought, we may want to keep it a pure variable. Because use of std::unique_ptr means we allocate it on a heap. I know the size of `QUICFrameInformation` is over 1KB now, so the copy cost is not so small, but I think we can reduce the size, perhaps less than 128 bytes.

If we really want to allocate a memory for QUICFrameInformation, we should probably use ATS's memory allocator.

Reply:
Yep, I agree.

Reply:
Hmm, I want to avoid keeping frames. The original goal of this refactoring is to not use allocators for frames.